app/controllers/users/sessions_controller.rb

```diff
@@ -128,7 +128,7 @@ def cache_active_profile
       cacher = Pii::Cacher.new(current_user, user_session)
       profile = current_user.decorate.active_or_pending_profile
       begin
-        cacher.save(auth_params[:password], profile)
+        cacher.save(auth_params[:password], profile) if profile.present?
```

Comment:
@jmhooper This is what I finally came to. As far as I can tell, the test that was failing (`spec/features/users/sign_in_spec.rb` line 244) was testing that the email attribute couldn't be decrypted and didn't have anything to do with this line. When I ran the test against the code in `master`, the user didn't have any profile and this method never got called with this test.

Reply:
This might be due to factory_bot stuff. I ran into a similar situation in the follow-on PR and found a fix. I'll test it later on this PR and roll it in here if it lets me get rid of this particular change.

Reply:
Yeah, I can totally believe it is something weird with factory bot.

Reply:
I'm looking at this and can't seem to figure out how it works at all without this conditional there.